lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -239,6 +240,13 @@ static ParseResult parseComponentOp(OpAsmParser &parser,
   if (body->empty())
     body->push_back(new Block());
 
+  // Combinational dialect used for assignment guards.
+  result.getContext()->loadDialect<comb::CombDialect>();
```

Comment:
I don't think this is the right way to do this. You're loading these dialects because the new pass might create them, right? It doesn't really make sense to load them as soon as a ComponentOp is parsed, because who knows, that pass might not be run.

There is a mechanism to handle exactly this: https://mlir.llvm.org/docs/PassManagement/#dependent-dialects. Passes that may produce ops in other dialect(s) have a way to indicate those dialects. For example: https://github.com/llvm/circt/blob/main/include/circt/Conversion/Passes.td#L28

Reply:
Beautiful, thank you for pointing this out, as well as the respective documentation (which I was having trouble finding).